Dockerfile

```diff
@@ -1,8 +1,17 @@
 FROM ruby:2.3 as base
 
+ADD https://dl.yarnpkg.com/debian/pubkey.gpg /tmp/yarn-pubkey.gpg
+RUN apt-key add /tmp/yarn-pubkey.gpg && rm /tmp/yarn-pubkey.gpg
+RUN echo 'deb http://dl.yarnpkg.com/debian/ stable main' > /etc/apt/sources.list.d/yarn.list
+
+ADD https://deb.nodesource.com/setup_12.x /tmp/node-setup.sh
+RUN bash /tmp/node-setup.sh && rm /tmp/node-setup.sh
+
 RUN apt-get update && apt-get install -y \
     graphviz \
     imagemagick \
+    nodejs \
+    yarn \
```

Comment:
yarn and nodejs are only required for the test containers, right?

Therefore I suggest to put them into a separate build stage. (See [multistage build](https://docs.docker.com/develop/develop-images/multistage-build/) ).

```Dockerfile
# ... existing dockerfile without changes ...

####################################################################
FROM base as test

# install nodejs and yarn
RUN wget -qO - https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add - \
  && echo 'deb http://dl.yarnpkg.com/debian/ stable main' > /etc/apt/sources.list.d/yarn.list \
  && wget -qO - https://deb.nodesource.com/setup_12.x | bash \
  && apt-get update && apt-get install -y \
    nodejs \
    yarn \
 && rm -rf /var/lib/apt/lists/*

ENV RAILS_ENV=test

####################################################################
# This should be the last build stage in the file. A `docker build`
# without args will tag this stage as the final image.
FROM base
```

Reply:
Btw: The `docker-compose.yml` already has the `test` stage configured as build target, that's why this should work :)

Reply:
I'm not sure, why we even install it. I think the only reason is if we wanted to run cypress headless in a docker container. But this wouldn't work, as we had to start a testserver first and at least have a separate service for the tests. The intention right now is to run the e2e tests outside of docker.
It is possible that I needed node/yarn in my first tests while installing cypress-on-rails.

I'm going to remove that part for now.